dev_guide/getting_traffic_into_cluster.adoc

```diff
@@ -0,0 +1,179 @@
+
+There are load balancers available on
+xref:../install_config/configuring_aws.adoc#install-config-configuring-aws[AWS] and
+xref:../install_config/configuring_gce.adoc#install-config-configuring-gce[GCE] clouds as well as
+xref:../admin_guide/tcp_ingress_external_ports.adoc#admin-guide-expose-external-ports[non-cloud].
+
+The non-cloud
+xref:../admin_guide/tcp_ingress_external_ports.adoc#admin-guide-expose-external-ports[LoadBalancer] allocates
+a unique IP from a configured pool. This limits you to a single ingress IP
+(can be a VIP, but still will be a single machine for initial load balancing).
+It simplifies the admin job by providing the needed IP address, but uses one
+IP per service.
+
+Load balancing is described
+link:http://kubernetes.io/docs/user-guide/services/#type-loadbalancer[here].
+
+[[using-externalIP]]
+== Using a Service ExternalIP
+
+The user can manually assign
+link:http://kubernetes.io/docs/user-guide/services/#external-ips[ExternalIPs]
+to the service. The supplied list of IP addresses are used for load balancing
+incoming requests. The service port is opened on all nodes running kube-proxy.
+Doing this requires elevated permissions to assign, and manual tracking of the
+IP:ports that are in use.
+
+An externally visible IP can be configured in one of several ways:
+
+- manually configure the ExternalIP with a known external IP address.
+- configure ExternalIP to a
+xref:../admin_guide/high_availability.adoc#admin-guide-high-availability[VIP]
+that is generated from (VRRP)
+- In a cloud (xref:../install_config/configuring_aws.adoc#install-config-configuring-aws[AWS] or
+xref:../install_config/configuring_gce.adoc#install-config-configuring-gce[GCE])
+use type=LoadBalancer
+- In a non-cloud environment, configure an ingressIP range (IngressIPNetworkCIDR),
+service.type=LoadBalancer and service.port.ingressIP
+
+The admin must make sure the external IPs are routed to the nodes and local
+firewall rules on all nodes allow access to the open port.
+
+Use the same nodes as the router but advertise them with VRRP and use
+the DNS address of the router nodes.
+
+
+[[using-nodeport]]
+== Using a Service NodePort
+
+Use NodePorts to expose the service nodePort on all nodes in the cluster.
+NodePorts are in the 30-60k range by default, which means a NodePort is
+unlikely to match a service's intended port (e.g., 8080 might be exposed
+as 31020).  This use of ports is wasteful of scarce port resources.
+However, it is a little easier to set up. Again, this requires more privilege.
+
+The admin must make sure the external IPs are routed to the nodes and local
+firewall rules on all nodes allow access to the open port.
+
+NodePorts and externalIP are independent and both can be used concurrently.
+
+[[virtual-ip]]
+== Using Virtual IPs
+
+All of the approaches can expose
+xref:../admin_guide/high_availability.adoc#admin-guide-high-availability[VIPs] (virtual IPs). The Virtual
+IPs are generated from the VRRP protocol.
```

Comment:
@marun I made some changes but I think it could use better wording. Could you take another look?
Thanks, phil